server/apiSchemas/compileRequest.json

```diff
@@ -0,0 +1,4 @@
+{
+    "language": { "type": "string", "enum": ["c", "c++", "rust"] },
```

Comment:
this is just a basic schema so that we have an idea - @slmaturen if you named the request fields differently on the back end i'm happy to change mine or think up better naming here

Reply:
I really like the idea of an enum list there. Makes a lot of sense in terms of our languages. That naming might help in other places as well.

Reply:
With the file upload, the API is currently detecting language type based on the extension but this does match the types the API is using! Works for me!